### PR TITLE
RUN-2860 Increase waiting time for check period

### PR DESCRIPTION
**Pull request by carlosrfranco**
**Is this a bugfix, or an enhancement? Please describe.**
Try to fix a flaky test increasing waiting time to check period to avoid hibernate exception when trying to access an object that could be removed

**Describe the solution you've implemented**

**Describe alternatives you've considered**

**Additional context**
Checking log from circleci, we can see this error when get the executions from api request:
`[docker-java-stream-2080914965] INFO org.rundeck.util.container.RdContainer - STDOUT: org.hibernate.ObjectNotFoundException: No row with the given identifier exists: [rundeck.Workflow#22]
    [docker-java-stream-2080914965] INFO org.rundeck.util.container.RdContainer - STDOUT:       at org.hibernate.boot.internal.StandardEntityNotFoundDelegate.handleEntityNotFound(StandardEntityNotFoundDelegate.java:28) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
    [docker-java-stream-2080914965] INFO org.rundeck.util.container.RdContainer - STDOUT:       at org.hibernate.proxy.AbstractLazyInitializer.checkTargetState(AbstractLazyInitializer.java:298) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
    [docker-java-stream-2080914965] INFO org.rundeck.util.container.RdContainer - STDOUT:       at org.hibernate.proxy.AbstractLazyInitializer.initialize(AbstractLazyInitializer.java:187) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
    [docker-java-stream-2080914965] INFO org.rundeck.util.container.RdContainer - STDOUT:       at org.hibernate.proxy.AbstractLazyInitializer.getImplementation(AbstractLazyInitializer.java:322) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
    [docker-java-stream-2080914965] INFO org.rundeck.util.container.RdContainer - STDOUT:       at org.hibernate.proxy.pojo.bytebuddy.ByteBuddyInterceptor.intercept(ByteBuddyInterceptor.java:45) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
    [docker-java-stream-2080914965] INFO org.rundeck.util.container.RdContainer - STDOUT:       at org.hibernate.proxy.ProxyConfiguration$InterceptorDispatcher.intercept(ProxyConfiguration.java:95) ~[hibernate-core-5.6.15.Final.jar!/:5.6.15.Final]
    [docker-java-stream-2080914965] INFO org.rundeck.util.container.RdContainer - STDOUT:       at rundeck.data.util.ExecReportUtil.summarizeJob(ExecReportUtil.groovy:58) ~[grails-rundeck-data-shared-0.1-plain.jar!/:?]
    [docker-java-stream-2080914965] INFO org.rundeck.util.container.RdContainer - STDOUT:       at rundeck.services.ExecutionService$__tt__respondExecutionsJson_closure100.doCall(ExecutionService.groovy:281) ~[classes!/:?]
    [docker-java-stream-2080914965] INFO org.rundeck.util.container.RdContainer - STDOUT:       at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
    [docker-java-stream-2080914965] INFO org.rundeck.util.container.RdContainer - STDOUT:       at rundeck.services.ExecutionService.$tt__respondExecutionsJson(ExecutionService.groovy:275) ~[classes!/:?]
    [docker-java-stream-2080914965] INFO org.rundeck.util.container.RdContainer - STDOUT:       at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
    [docker-java-stream-2080914965] INFO org.rundeck.util.container.RdContainer - STDOUT:       at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]`

### CODE REVIEW

**Comment by mrdubr** [comment]
Hey @carlosrfranco , Please, move it to the "Ready for review" if it's ready for one.

File: functional-test/src/test/groovy/org/rundeck/tests/functional/api/execution/ExecutionSpec.groovy [hunk 1]
```diff
@@ -134,7 +134,7 @@ class ExecutionSpec extends BaseContainer {
             // Waits for all executions to get cleaned
             waitFor(ExecutionUtils.Retrievers.executionsForProject(client, projectName),
                 {it.isEmpty()},
-                WaitingTime.XTRA_EXCESSIVE)
+                WaitingTime.XTRA_EXCESSIVE, WaitingTime.MODERATE)
```
Comment (mrdubr) [inline functional-test/src/test/groovy/org/rundeck/tests/functional/api/execution/ExecutionSpec.groovy:137]: It's not obvious to me how checking for something less frequently (`WaitingTime.MODERATE`) can fix this race condition in a *reliable manner*. IMO, a proper fix should handle this error condition and interpret it appropriately for the `waitFor` "acceptance" logic.

**Review by mrdubr — CHANGES_REQUESTED** [review]
Please, see the comment

File: functional-test/src/test/groovy/org/rundeck/tests/functional/api/execution/ExecutionSpec.groovy (hunk 1 above)
Comment (carlosrfranco) [reply functional-test/src/test/groovy/org/rundeck/tests/functional/api/execution/ExecutionSpec.groovy:137]: Thanks for reviewing and I will double check it. What I could see is that this request make some queries to get executions data before responding it to the client side. On this test, this request is made each second and, in parallel, the server is performing an executions clean up in the project. So, for a one second period check, there is a big chance that some executions to be deleted after a request start and before it ends. I think there is no reason to have a check period for each one second, and increasing it to 5 seconds could be reasonable

Comment (mrdubr) [reply functional-test/src/test/groovy/org/rundeck/tests/functional/api/execution/ExecutionSpec.groovy:137]: Changing the timing of the "check" can decrease the likelihood of the flake, but does not address the root cause.
For the reliable fix, you need to handle the error condition.

What error are you getting from the API? A 4xx or a 5xx?

File: functional-test/src/test/groovy/org/rundeck/tests/functional/api/execution/ExecutionSpec.groovy [hunk 2]
```diff
@@ -131,10 +133,19 @@ class ExecutionSpec extends BaseContainer {
                 json2.executions.size() == 6
             }
         cleanup:
+            def executionsRetrived = {->
```
Comment (mrdubr) [inline functional-test/src/test/groovy/org/rundeck/tests/functional/api/execution/ExecutionSpec.groovy:136]: When naming closures, it's generally a good practice to use verbs, as this makes it clear that the closure represents an action or behavior. This follows the same principle as naming methods or functions, where verbs are typically used to indicate what the method or function does.

File: functional-test/src/test/groovy/org/rundeck/tests/functional/api/execution/ExecutionSpec.groovy [hunk 3]
```diff
@@ -131,10 +133,19 @@ class ExecutionSpec extends BaseContainer {
...
+                try {
+                    return ExecutionUtils.Retrievers.executionsForProject(client, projectName).get()
+                } catch (JsonParseException e) {
+                    // if request doesnt return an expected json, return null to retry
```
Comment (mrdubr) [inline functional-test/src/test/groovy/org/rundeck/tests/functional/api/execution/ExecutionSpec.groovy:140]: You should not be discarding the `e`.
The least thing you can do it to log it.

**Review by mrdubr — APPROVED** [review]
Great fix!
I minor soft nit.
Approved, conditional on the exception logging.

**Pull request merged**